src/Faithlife.Utility/DirectoryUtility.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace Faithlife.Utility
+{
+	/// <summary>
+	/// Provides utility methods for working with directories.
+	/// </summary>
+	public static class DirectoryUtility
+	{
+		/// <summary>
+		/// Recursively finds all files matching <paramref name="searchPatterns"/> in the folders specified by
+		/// <paramref name="directoryPaths"/> and their subfolders.
+		/// </summary>
+		/// <param name="directoryPaths">The directory paths to search.</param>
+		/// <param name="searchPatterns">The search patterns to match against the names of files.</param>
+		/// <param name="cancellationToken">The cancellation token.</param>
+		/// <returns>A sequence of fully-qualified file paths.</returns>
+		public static IEnumerable<string> FindFiles(IEnumerable<string> directoryPaths, IEnumerable<string> searchPatterns, CancellationToken cancellationToken)
+		{
+			return DoFindFiles(directoryPaths, searchPatterns, DoFindFiles, Directory.GetDirectories, cancellationToken);
```

Comment:
Shouldn't we use `Directory.EnumerateDirectories`? And allow cancellation as we enumerate?